### PR TITLE
Add request parameter 'cluster_manager_timeout' and deprecate 'master_timeout' - in Index APIs except index template APIs

### PR DESCRIPTION
**Pull request by tlfeng**
### Description

- Deprecate the request parameter `master_timeout` that used in Index APIs which have got the parameter. 
(except index template APIs, which is addressed in PR https://github.com/opensearch-project/OpenSearch/pull/2678) 
- Add alternative new request parameter `cluster_manager_timeout`.
- Add unit tests.

List of the `Index` APIs that have got request parameter `master_timeout`:
- Index management:
Create index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/create-index/
Delete index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/delete-index/
Get index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/get-index/
Open index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/open-index/
Close index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/close-index/
Clone index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/clone/
Shrink index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/shrink-index/
Split index API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/split/
Rollover index API `POST /<rollover-target>/_rollover/<target-index>`
- Alias management:
Add index alias API https://opensearch.org/docs/latest/opensearch/rest-api/alias/
Delete index alias API `DELETE /<index>/_alias/<alias>`
- Mapping management:
Get mapping API
Put mapping API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/put-mapping/
- Index settings:
Get Index Settings API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/get-settings/
Update index settings API https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/update-settings/
- Index templates:
Get (Legacy) Index Template API `GET /_template/<index-template>`
Delete (Legacy) Index Template API `DELETE /_template/<index-template>`
Put (Legacy) Index Template API `PUT /_template/<index-template>`
Get (Composable) Index Template API `GET /_index_template/<index-template>`
Delete (Composable) Index Template API
Put (Composable) Index Template API
Get Component Template API `GET /_component_template/<index-template>`
Delete Component Template API
Put Component Template API
Simulate index API `POST /_index_template/_simulate_index/{name}`
Simulate index template API `POST /_index_template/_simulate`
- Dangling indices:
Delete Dangling Index API
Import Dangling Index API `POST /_dangling/<index-uuid>`
- Other:
Add index block API `PUT /<index>/_block/<block>`

### Issues Resolved
A part of issue https://github.com/opensearch-project/OpenSearch/issues/2511
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by nknize — CHANGES_REQUESTED** [review]
Same change request as in #2658

File: server/src/main/java/org/opensearch/rest/action/admin/indices/RestCloseIndexAction.java [hunk 1]
```diff
@@ -73,4 +79,19 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
...
     }
 
+    /**
+     * Parse the deprecated request parameter 'master_timeout', and add deprecated log if the parameter is used.
+     * It also validates whether the value of 'master_timeout' is the same with 'cluster_manager_timeout'.
+     * Remove the method along with MASTER_ROLE.
+     * @deprecated As of 2.0, because promoting inclusive language.
+     */
+    @Deprecated
+    private void parseDeprecatedMasterTimeoutParameter(CloseIndexRequest closeIndexRequest, RestRequest request) {
```
[the remark this answers is not in the document]
Comment (tlfeng) [reply server/src/main/java/org/opensearch/rest/action/admin/indices/RestCloseIndexAction.java:89]: @nknize Thanks for reviewing this PR! I applied your suggested change in a new PR https://github.com/opensearch-project/OpenSearch/pull/2670. (I will continue this conversation in PR https://github.com/opensearch-project/OpenSearch/pull/2658#pullrequestreview-926303776)

[the remark this answers is not in the document]
Comment (tlfeng) [reply server/src/main/java/org/opensearch/rest/action/admin/indices/RestCloseIndexAction.java]: The suggested change is merged with PR #2670, and I have updated this PR.

**Comment by tlfeng** [comment]
In log 4016
```
> Task :client:rest-high-level:asyncIntegTest

REPRODUCE WITH: ./gradlew ':client:rest-high-level:asyncIntegTest' --tests "org.opensearch.client.documentation.IndicesClientDocumentationIT.testDeleteAlias" -Dtests.seed=DADB4A24C4FA8AC3 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=it-CH -Dtests.timezone=Asia/Kuching -Druntime.java=17

org.opensearch.client.documentation.IndicesClientDocumentationIT > testDeleteAlias FAILED
    org.opensearch.client.WarningFailureException: method [DELETE], host [http://127.0.0.1:32983], URI [/index1/_alias/alias1?master_timeout=1m&timeout=2m], status line [HTTP/1.1 200 OK]
    Warnings: [Deprecated parameter [master_timeout] used. To promote inclusive language, please use [cluster_manager_timeout] instead. It will be unsupported in a future major version.]
    {"acknowledged":true}
        at __randomizedtesting.SeedInfo.seed([DADB4A24C4FA8AC3:3560A5B72A0A4E44]:0)
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:346)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:320)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:295)
        at app//org.opensearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1762)
        at app//org.opensearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1745)
        at app//org.opensearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1709)
        at app//org.opensearch.client.IndicesClient.deleteAlias(IndicesClient.java:1785)
        at app//org.opensearch.client.documentation.IndicesClientDocumentationIT.testDeleteAlias(IndicesClientDocumentationIT.java:2981)
```

**Comment by tlfeng** [comment]
In log 4017
```
* What went wrong:
Execution failed for task ':qa:mixed-cluster:v2.1.0#mixedClusterTest'.
> `cluster{:qa:mixed-cluster:v2.1.0}` failed to wait for cluster health yellow after 40 SECONDS
    IO error while waiting cluster
    503 Service Unavailable
```

**Comment by tlfeng** [comment]
In log 4054:
```
Execution failed for task ':client:rest-high-level:asyncIntegTest'.

Tests with failures:
 - org.opensearch.client.BulkProcessorIT.testGlobalParametersAndSingleRequest
 - org.opensearch.client.IndicesClientIT.testUpdateAliases
 - org.opensearch.client.IndicesClientIT.testClone
 - org.opensearch.client.IndicesClientIT.testShrink
 - org.opensearch.client.IndicesClientIT.testDeleteAlias
 - org.opensearch.client.IndicesClientIT.testGetFieldMapping
 - org.opensearch.client.IndicesClientIT.testSplit
 - org.opensearch.client.IndicesClientIT.testRollover
 - org.opensearch.client.IndicesClientIT.testGetMapping
 - org.opensearch.client.IndicesClientIT.testOpenExistingIndex
 - org.opensearch.client.IndicesClientIT.testDeleteIndex

291 tests completed, 11 failed, 1 skipped
```

**Comment by tlfeng** [comment]
In log 4153:
```
REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v2.0.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=tasks.list/10_basic/tasks_list test}" -Dtests.seed=B4F39ACD8A2A3DED -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-PE -Dtests.timezone=Australia/ACT -Druntime.java=17

org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=tasks.list/10_basic/tasks_list test} FAILED
    java.lang.AssertionError: Failure at [tasks.list/10_basic:15]: field [nodes.$node_id.roles] doesn't have a true value
    Expected: not null
         but: was null
        at __randomizedtesting.SeedInfo.seed([B4F39ACD8A2A3DED:3CA7A51724D65015]:0)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.executeSection(OpenSearchClientYamlSuiteTestCase.java:454)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.test(OpenSearchClientYamlSuiteTestCase.java:427)
```
It's reproducible locally.

**Review by nknize — APPROVED** [review]
Oh yeah.. this is much cleaner!!! Nice work.

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.get_template.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
       },
       "master_timeout":{
         "type":"time",
-        "description":"Explicit operation timeout for connection to master node"
+        "description":"Specify timeout for connection to master"
```
Comment (tlfeng) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.get_template.json:37]: I forgot to deprecate the parameter here..

Comment (tlfeng) [reply rest-api-spec/src/main/resources/rest-api-spec/api/indices.get_template.json]: Updated in commit https://github.com/opensearch-project/OpenSearch/pull/2660/commits/b914741f9683edc583b7ca6ebc0b5615db08d248

**Comment by nknize** [comment]
#1703 strikes again!

**Comment by nknize** [comment]
start gradle check

**Comment by tlfeng** [comment]
In log 4188:
```
* What went wrong:
Execution failed for task ':modules:analysis-common:test'.
> Process 'Gradle Test Executor 446' finished with non-zero exit value 1
  This problem might be caused by incorrect test process configuration.
  Please refer to the test execution section in the User Manual at https://docs.gradle.org/7.4.2/userguide/java_testing.html#sec:test_execution
```
start gradle check

**Pull request merged**